docs/keps/20200504-cleanup-fed-resources.md

```diff
@@ -0,0 +1,113 @@
+---
+kep-number: 0
+short-desc: Cleanup federated resource on unjoin
+title: Cleanup Federated Resource
+authors:
+  - "@jr0d"
+reviewers:
+  - "@jimmidyson"
+  - "@hectorj2f"
+approvers:
+  - "@jimmidyson"
+editor: TBD
+creation-date: 2020-05-18
+last-updated: 2020-05-04
+status: provisional
+---
+
+# Cleanup federated resources
+
+## Table of Contents
+
+* [Cleanup Federated Resources](#cleanup-federated-resources)
+  * [Table of Contents](#table-of-contents)
+  * [Summary](#summary)
+  * [Motivation](#motivation)
+    * [User Story](#user-stroy)
+    * [Goals](#goals)
+  * [Proposals](#proposal)
+    * [Implementation Details](#implementation-details)
+      * [Best Effort Strategy](#best-effort-strategy)
+      * [Required Strategy](#required-strategy)
+    * [Addition ClusterConditionTypes (optional)](#additional-clusterconfigtypes-optional)
+      * [Unjoining ClusterConfigurationType](#unjoining-clusterconfigurationtype)
+      * [Failed ClusterConfigurationType](#failed-clusterconfigurationtype)
+  * [Risks](#risks)
+    * [Adding Cluster Conditions](#adding-cluster-conditions)
+
+This document proposes a mechanism to specify that a federated resource should be removed from a managed cluster when leaving federation.
+
+## Motivation
+
+In practice, federation is often used to join clusters to some higher level centralized management plane, identity management for example. To achieve this, federation is used to propagate a configuration state to clusters which overrides an existing configuration state. When a cluster is unjoined, the federated configuration becomes invalid requiring the operator to manually remove the conflicting resources.
```

Comment:
I think you should use `kubefed` as the disambiguated name and not `federation`. We did update the term `federation` to `kubernetes cluster federation` and thats what is reflected across the documentation and code in this repo.
So it would be a `federated resource` or `federated cluster` but `kubefed control plane` or `join kubefed`, etc.
This applies to the whole document.